core/src/main/java/bisq/core/dao/governance/voteresult/MissingDataRequestService.java

```diff
@@ -51,4 +76,39 @@ public void start() {
     public void sendRepublishRequest() {
         republishGovernanceDataHandler.sendRepublishRequest();
     }
+
+    public void reRepublishAllGovernanceData() {
+        ObservableList<ProposalPayload> proposalPayloads = proposalService.getProposalPayloads();
+        proposalPayloads.forEach(proposalPayload -> {
+            // We want a random delay between 0.1 and 30 sec. depending on the number of items
+            int delay = Math.max(100, Math.min(30_000, new Random().nextInt(proposalPayloads.size() * 500)));
+            UserThread.runAfter(() -> {
+                boolean success = p2PService.addPersistableNetworkPayload(proposalPayload, true);
+                String txId = proposalPayload.getProposal().getTxId();
+                if (success) {
+                    log.warn("We received a RepublishGovernanceDataRequest and re-published a proposalPayload to " +
```

Comment:
Yes was also wondering. But it make it more visible if that occurres so that anyone can see if any peer is missing data. Basically the P2P network should be reliable enough to not cause that.